grakn-graph/src/main/java/ai/grakn/graph/internal/OntologyElementImpl.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graph.internal;
+
+import ai.grakn.concept.EntityType;
+import ai.grakn.concept.OntologyElement;
+import ai.grakn.concept.RelationType;
+import ai.grakn.concept.RoleType;
+import ai.grakn.concept.TypeId;
+import ai.grakn.concept.TypeLabel;
+import ai.grakn.exception.GraphOperationException;
+import ai.grakn.util.Schema;
+import org.apache.tinkerpop.gremlin.structure.Direction;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ *     Ontology or Schema Specific Element
+ * </p>
+ *
+ * <p>
+ *     Allows you to create schema or ontological elements.
+ *     These differ from normal graph constructs in two ways:
+ *     1. They have a unique {@link TypeLabel} which identifies them
+ *     2. You can link them together into a hierarchical structure
+ * </p>
+ *
+ * @author fppt
+ *
+ * @param <T> The leaf interface of the object concept.
+ *           For example an {@link EntityType} or {@link RelationType} or {@link RoleType}
+ */
```

Comment:
Is it worth repeating these docs for the implementation? To me it seems like one more place for the docs to be inconsistent.